cpp/src/neighbors/hnsw_c.cpp

```diff
@@ -148,9 +148,12 @@ extern "C" cuvsError_t cuvsHnswIndexDestroy(cuvsHnswIndex_t index_c_ptr)
   return cuvs::core::translate_exceptions([=] {
     auto index = *index_c_ptr;
 
-    if (index.dtype.code == kDLFloat) {
+    if (index.dtype.code == kDLFloat && index.dtype.bits == 32) {
       auto index_ptr = reinterpret_cast<cuvs::neighbors::hnsw::index<float>*>(index.addr);
       delete index_ptr;
+    } else if (index.dtype.code == kDLFloat && index.dtype.bits == 16) {
+      auto index_ptr = reinterpret_cast<cuvs::neighbors::hnsw::index<half>*>(index.addr);
+      delete index_ptr;
     } else if (index.dtype.code == kDLInt) {
```

Comment:
Should this and the other conditionals below check for number of bits?

Reply:
Yes I added a check to enforce it